Flesh out rest of print_stack_of

pnkfelix opened the pull request.

pnkfelix committed:
make addresses without stacks still respect address normalization.

pnkfelix committed:
flesh out the rest of print_stack_of.

pnkfelix commented:
r? @bryangarza

bryangarza commented on krabcake/rs_hello/src/lib.rs:
```diff
@@ -470,10 +470,23 @@ pub extern "C" fn rs_client_request_print_stack_of(
+                }
+                match item {
+                    Item::Unique(tag) => {
+                        vgPlain_umsg(b"Unique(%lld)\0".as_ptr() as *const c_char, tag.0);
```
You can actually derive `Debug` on Stack and all its nested types, and you can avoid having to manually write this whole loop

pnkfelix replied:
I took the best part of this feedback and turned it into commit https://github.com/pnkfelix/krabcake-vg/pull/27/commits/834564206df8b9cb4cb30f9bee065aa0b1e19d88

pnkfelix committed:
Review feedback: We can use Rust's formatting machinery rather than hard code every call to vgPlain_umsg.

The actual review feedback has suggested using `fmt::Debug` here, but I opted
not to do that, because I intepret these umsg calls as being part of the
user-visible output from the tool, and for that, `fmt::Display` is the
appropriate trait, not `fmt::Debug`. (This is a shame, because there are no
great ways (at least not in libstd) to recursively print list/set/map
collections using the elements' `fmt::Display` implementations. But I am not
going to try to tackle that problem today; I just wrote the standard code to
iterate over the list.

The thing that *was* fun about this code: I added `VgPlainUmsgWriter`, which is
a `fmt::Writer` implementation that emits output via `vgPlain_usmg`. This does
simplify things nicely, and I suspect we can use it to clean other other parts
of the code base.

pnkfelix committed:
added note for potential future change to this method's protocol.

bryangarza reviewed (approved):
LGTM once CI is passing

pnkfelix commented on krabcake/rs_hello/src/lib.rs:
```diff
@@ -448,13 +449,44 @@ pub extern "C" fn rs_client_request_print_tag_of(
+            follow[0] = b'\0';
+            // now print the chunk, with the copied bytes, out.
+            unsafe {
+                vgPlain_umsg("%s\0".as_ptr() as *const c_char, buf.as_ptr());
```
It might be good to have a test (in the outer `kc` crate that has the test suite) that exercises what  is special about this line.

Namely, the reason we pass "%s\0" instead of just passing in `buf` directly to `vgPlain_usmg` is to handle the case where `buf` itself contains printf specifiers like `%s` or `%d` etc.

pnkfelix replied:
This is https://github.com/pnkfelix/krabcake/pull/23

pnkfelix committed:
appease rustfmt gods.
